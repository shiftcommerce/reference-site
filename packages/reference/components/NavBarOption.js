import { Component } from 'react'
import Link from 'next/link'
import classNames from 'classnames'

class NavBarOption extends Component {
  render() {
    let {
      link,
      title,
      index
    } = this.props

    return <Link href={ link }>
      <a className={ classNames('o-nav__option', { 'o-nav__option--first': index === 0 }) } >
        { title }
      </a>
    </Link>
  }
}

export default NavBarOption