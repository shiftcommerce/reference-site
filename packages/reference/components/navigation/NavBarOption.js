// Libraries
import { Component } from 'react'
import Link from 'next/link'
import classNames from 'classnames'

class NavBarOption extends Component {
  render () {
    let {
      href,
      as,
      title,
      index
    } = this.props

    return (
      <Link href={href} as={as}>
        <a className={classNames('o-nav__option', { 'o-nav__option--first': index === 0 })} >
          { title }
        </a>
      </Link>
    )
  }
}

export default NavBarOption
