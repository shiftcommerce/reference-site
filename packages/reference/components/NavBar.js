// Libraries
import { Component } from 'react'

// Objects
import NavBarOption from './NavBarOption'

class NavBar extends Component {
  renderNavOptions () {
    const navOptions = [
      { title: 'Mens', link: '/categories/mens' },
      { title: 'Womens', link: '/womens' },
      { title: 'Bathroom', link: '/categories/Bathroom' },
      { title: 'Girls', link: '/girls' },
      { title: 'Other', link: '/other' },
      { title: 'Offers', link: '/offers' }
    ]

    return navOptions.map((navOption, index) =>
      <NavBarOption key={index} index={index} title={navOption.title} link={navOption.link} />
    )
  }

  render () {
    return (
      <div className='o-nav' role='navigation' >
        { this.renderNavOptions() }
      </div>
    )
  }
}

export default NavBar
