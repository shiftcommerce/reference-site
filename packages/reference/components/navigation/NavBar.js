// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'

// Objects
import NavBarOption from './NavBarOption'

class NavBar extends Component {
  renderNavOptions (categories) {
    return (
      categories.data.map((category, index) =>
        <NavBarOption key={index} index={index} title={category.title} href={`/categories/category?id=${category.id}`} as={`/categories/${category.id}`} />
      )
    )
  }

  render () {
    const {
      categories
    } = this.props

    if (categories.loading) {
      return (
        <p>Loading...</p>
      )
    } else if (categories.error) {
      return (
        <p style={{'textAlign': 'center'}} >
          Sorry! There is an error in loading menus
          {categories.error}
        </p>
      )
    } else {
      return (
        <div className='o-nav' role='navigation' >
          { this.renderNavOptions(categories) }
        </div>
      )
    }
  }
}

const mapStateToProps = (state) => {
  return {
    categories: state.categories || []
  }
}

export default connect(mapStateToProps)(NavBar)
