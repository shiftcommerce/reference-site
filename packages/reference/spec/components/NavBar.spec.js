import NavBar from '../../components/NavBar'
import NavBarOption from '../../components/NavBarOption'

test('renders the nav bar options', () => {
  // arrange

  // act
  const wrapper = shallow(
    <NavBar />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toContainReact(<NavBarOption index={0} title='Mens' link='/categories/mens' />)
  expect(wrapper).toContainReact(<NavBarOption index={1} title='Womens' link='/womens' />)
  expect(wrapper).toContainReact(<NavBarOption index={2} title='Bathroom' link='/categories/Bathroom' />)
  expect(wrapper).toContainReact(<NavBarOption index={3} title='Girls' link='/girls' />)
  expect(wrapper).toContainReact(<NavBarOption index={4} title='Other' link='/other' />)
  expect(wrapper).toContainReact(<NavBarOption index={5} title='Offers' link='/offers' />)
})
