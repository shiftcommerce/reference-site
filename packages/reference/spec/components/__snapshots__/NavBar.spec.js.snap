// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the nav bar options 1`] = `
<div
  className="o-nav"
  role="navigation"
>
  <NavBarOption
    index={0}
    link="/categories/mens"
    title="Mens"
  />
  <NavBarOption
    index={1}
    link="/womens"
    title="Womens"
  />
  <NavBarOption
    index={2}
    link="/categories/Bathroom"
    title="Bathroom"
  />
  <NavBarOption
    index={3}
    link="/girls"
    title="Girls"
  />
  <NavBarOption
    index={4}
    link="/other"
    title="Other"
  />
  <NavBarOption
    index={5}
    link="/offers"
    title="Offers"
  />
</div>
`;
